### ISSUE TITLE
How do you log loss and accuracy while using horovod.spark.torchestimator?

### ISSUE THREAD

**Issue by SRITANU** [horovod/horovod#2492]
We are trying to train a neural net on a spark dataframe on databricks azure cluster using horovod's spark [torchestimator](https://github.com/horovod/horovod/blob/master/horovod/spark/torch/estimator.py). We could train the model. 

But we'd like to log metrics like loss, accuracy per epochs for later use. Any idea how to do that? Where do I look at? I also looked at the estimator store which keeps all the intermediate files but found a bunch of parquet and pickle files.

**Comment by abditag2** [comment]
@SRITANU You can get the logs using the history field:
```
transformer = torch_estimator.fit(df, ...)
history = transformer.getHistory()
```

**Comment by SRITANU** [comment]
Thanks @abditag2 . That helped a lot. Another issue that I am facing is:

I am using a databricks azure cluster of 1 driver with 3 worker nodes with each having 4 gpus. Strangely, I could only use `num_procs=3` as an argument to the torchestimator. Any number greater than that I am getting a `mpi exited with code 1 error` . Also, while monitoring the `gpu_util` with ganglia, I could only see the gpu2 of the three workers being utilized. Not the `gpu0` or `gpu1` or `gpu3`. Can you please help me figure this out? Where I am going wrong?
